skimage/transform/steerable.py

```diff
@@ -0,0 +1,425 @@
+        """
+        Parameters
+        ----------
+        lodft : 2-D array
+            DFT matrix of the higher layer
+        log_rad, angle: 2-D array
+            Helper matrices to help create the DFT mask
+        Xrcos, Yrcos: 2-D array
+            Represents the desired filter
+        ht: int
+            Current level of the pyramid that are being built
+
+        Notes
+        -----
+        This method is called recursively until ht reaches 1.
+        """
+
+        if (len(coeff) == 1):
+            return np.fft.fftshift(np.fft.fft2(coeff[0][0]))
+
+        else:
+
+            Xrcos = Xrcos - 1
+
+            # ========================== Orientation residue===================
+            himask = _point_op(log_rad, Yrcos, Xrcos)
+
+            lutsize = 1024
+            Xcosn = np.pi * \
+                np.array(range(-(2 * lutsize + 1), (lutsize + 2))) / lutsize
+            order = self.n_bands - 1
+            const = np.power(2, 2 * order) * \
+                np.square(np.math.factorial(order)) / \
+                (self.n_bands * np.math.factorial(2 * order))
+
+            Ycosn = np.sqrt(const) * np.power(np.cos(Xcosn), order)
+
+            orientdft = np.zeros(coeff[0][0].shape)
+
+            for b in range(self.n_bands):
+                anglemask = _point_op(
+                    angle, Ycosn, Xcosn + np.pi * b / self.n_bands)
+                banddft = np.fft.fftshift(np.fft.fft2(coeff[0][b]))
+                orientdft = orientdft + \
+                    np.power(np.complex(0, 1), order) * \
+                    banddft * anglemask * himask
+
+            # ============== Lowpass component are upsampled and convoluted ===
+            dims = np.array(coeff[0][0].shape)
+
+            lostart = (np.ceil((dims + 0.5) / 2) -
+                       np.ceil((np.ceil((dims - 0.5) / 2) + 0.5) / 2))\
+                .astype(np.int32)
+
+            loend = lostart + np.ceil((dims - 0.5) / 2).astype(np.int32)
+
+            nlog_rad = log_rad[lostart[0]:loend[0], lostart[1]:loend[1]]
+            nangle = angle[lostart[0]:loend[0], lostart[1]:loend[1]]
+            YIrcos = np.sqrt(np.abs(1 - Yrcos * Yrcos))
+            lomask = _point_op(nlog_rad, YIrcos, Xrcos)
+
+            nresdft = self._recon_pyr_level(
+                coeff[1:], nlog_rad, Xrcos, Yrcos, nangle)
+
+            resdft = np.zeros(dims, 'complex')
+            resdft[lostart[0]:loend[0], lostart[1]:loend[1]] = nresdft * lomask
+
+            return resdft + orientdft
+
+    def recon_scf_pyramid(self, coeff):
+        """ Reconstruct the image from its Steerable decomposition.
+
+        Parameters
+        ----------
+        coeff : list of lists of numpy array
+            Subbands of Steerable decomposition,
+            stored as a list of 'height' sublists,
+            Sublists correspond to decreasing radius level in Steerable pyramid
+            The first sublist contains the high pass subband.
```

Comment:
First sublist or first element of a sublist?

Reply:
Sometimes for this kind of explanation I find it helpful to use sample indices: `coeff[i][j] contains the subband of orientation j at height i.`, for example.